tiatoolbox/models/engine/engine_abc.py

```diff
@@ -1,21 +1,610 @@
+        merge_predictions (bool):
+            Whether to merge the predictions to form a 2-dimensional
+            map. This is only applicable `patch_mode` is False in inference.
+        resolution (Resolution):
+            Resolution used for reading the image. Please see
+            :obj:`WSIReader` for details.
+        units (Units):
+            Units of resolution used for reading the image. Choose
+            from either `level`, `power` or `mpp`. Please see
+            :obj:`WSIReader` for details.
+        patch_input_shape (tuple):
+            Size of patches input to the model. Patches are at
+            requested read resolution, not with respect to level 0,
+            and must be positive.
+        stride_shape (tuple):
+            Stride used during WSI processing. Stride is
+            at requested read resolution, not with respect to
+            level 0, and must be positive. If not provided,
+            `stride_shape=patch_input_shape`.
+        batch_size (int):
+            Number of images fed into the model each time.
+        labels (list | None):
+                List of labels. Only a single label per image is supported.
+        device (str):
+            Select the device to run the model. Default is "cpu".
+        num_loader_workers (int):
+            Number of workers used in torch.utils.data.DataLoader.
+        verbose (bool):
+            Whether to output logging information.
+
+    Examples:
+        >>> # array of list of 2 image patches as input
+        >>> import numpy as np
+        >>> data = np.array([np.ndarray, np.ndarray])
+        >>> engine = EngineABC(pretrained_model="resnet18-kather100k")
+        >>> output = engine.run(data, patch_mode=True)
+
+        >>> # list of 2 image patch files as input
+        >>> data = ['path/img.png', 'path/img.png']
+        >>> engine = EngineABC(pretrained_model="resnet18-kather100k")
+        >>> output = engine.run(data, patch_mode=False)
+
+        >>> # list of 2 image files as input
+        >>> image = ['path/image1.png', 'path/image2.png']
+        >>> engine = EngineABC(pretraind_model="resnet18-kather100k")
+        >>> output = engine.run(image, patch_mode=False)
+
+        >>> # list of 2 wsi files as input
+        >>> wsi_file = ['path/wsi1.svs', 'path/wsi2.svs']
+        >>> engine = EngineABC(pretraind_model="resnet18-kather100k")
+        >>> output = engine.run(wsi_file, patch_mode=True)
 
-    def __init__(self) -> None:
+    """
+
+    def __init__(
+        self: EngineABC,
+        model: str | nn.Module,
+        batch_size: int = 8,
+        num_loader_workers: int = 0,
+        num_post_proc_workers: int = 0,
+        weights: str | Path | None = None,
+        *,
+        device: str = "cpu",
+        verbose: bool = False,
+    ) -> None:
         """Initialize Engine."""
         super().__init__()
 
+        self.masks = None
+        self.images = None
+        self.patch_mode = None
```

Comment:
```suggestion
        self.patch_mode: bool = None
```
some missing type hints here